scripts/update-schema-latest.sh

```diff
@@ -0,0 +1,354 @@
+    if ! expr "${version}" : '[0-9][0-9]*\.[0-9][0-9]*$' >/dev/null; then
+        __error_exit "The format of the version specified by the -v option is invalid."
+    fi
+
+    # Extract the major number and the minor number.
+    __var_major=`echo ${version} | cut -f 1 -d .`
+    __var_minor=`echo ${version} | cut -f 2 -d .`
+
+    # If the major number is less than 3.
+    if [ ${__var_major} -lt 3 ]; then
+        __error_exit "This script does not support versions older than 3.0."
+    fi
+}
+
+
+#------------------------------------------------------------
+# __execute_command
+#------------------------------------------------------------
+__execute_command()
+{
+    command="$1"
+
+    # If the quiet mode is off.
+    if [ "${__var_quiet}" = "off" ]; then
+        # Print the command.
+        echo "${command}"
+    fi
+
+    # If the dry-run mode is off.
+    if [ "${__var_dry_run}" = "off" ]; then
+        # Execute the command.
+        sh -c "${command}"
+    fi
+}
+
+
+#------------------------------------------------------------
+# __checkout_gh_pages
+#------------------------------------------------------------
+__checkout_gh_pages()
+{
+    # The command to checkout the 'gh-pages' branch.
+    command="git checkout gh-pages"
+
+    # Execute the command.
+    __execute_command "${command}"
+}
+
+
+#------------------------------------------------------------
+# __update_schema
+#------------------------------------------------------------
+__update_schema()
+{
+    __update "schema"
+}
+
+
+#------------------------------------------------------------
+# __update_schema_base
+#------------------------------------------------------------
+__update_schema_base()
+{
+    __update "schema-base"
+}
+
+
+#------------------------------------------------------------
+# __update
+#------------------------------------------------------------
+__update()
+{
+    file_base="$1"
+
+    # The source file in the 'main' branch.
+    src_file="schemas/v${__var_version}/${file_base}.json"
+
+    # If the version is 3.0.
+    if [ ${__var_major} -eq 3 ] && [ ${__var_minor} -eq 0 ]; then
+        # Extract the 'YYYY-MM-DD' part of the "id" property.
+        date=`git show main:${src_file} | jq -rj '."id"|split("/")|last'`
+    else
+        # Extract the 'YYYY-MM-DD' part of the "$id" property.
+        date=`git show main:${src_file} | jq -rj '."$id"|split("/")|last'`
+    fi
+
+    # The destination directory in the 'gh-pages' branch.
+    dst_dir="oas/${__var_version}/${file_base}"
+
+    # The destination file in the 'gh-pages' branch.
+    dst_file="${dst_dir}/${date}"
+
+    # The command to copy the source file to the destination file.
+    copy_command="git show main:${src_file} > ${dst_file}"
+
+    # The command to update the 'latest' file.
+    link_command="cd ${dst_dir} && ln -sf ${date} latest"
+
+    # Execute the commands.
+    __execute_command "${copy_command}"
+    __execute_command "${link_command}"
```

Comment:
If #4034 is approved, this needs to become

```suggestion
    # The commands to update the 'latest' file.
    link_command="cd ${dst_dir} && ln -sf ${date} latest"
    link_command2="cd ${dst_dir} && ln -sf ${date} latest.json"

    # Execute the commands.
    __execute_command "${copy_command}"
    __execute_command "${link_command}"
    __execute_command "${link_command2}"
```